examples/get_matching/server.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_ext.h"
+
+#include "get_me.h"
+
+/*
+ * this file represent a server preparing messages which can be get from a client
+ *
+ * Usage: ./server [max messages possible to get]
+ *
+ * The server will exit after the specified number of messages have been got.
+ * By default, the server wait forever.
+ */
+
+/* used to perform a MEAppend*/
+int add_me(ptl_handle_ni_t nih, ptl_index_t pti, ptl_me_t me, ptl_handle_me_t* meh,
+	    ptl_handle_eq_t eqh, ptl_event_t ev, char *msg)

```

Comment:
Maybe we could replace the `msg` argument with a local array in the stack of this function, this unclutters a bit the call site